### DISCUSSION TITLE
Access Model in Custom Field on Create Page

### DISCUSSION THREAD

**Discussion by intrepidws** [filamentphp/filament#3687] [Help]
I have a custom field that needs to be able to access the model itself for a few reasons. Using `$this->record` works fine on an edit page, but this doesn't work on a create page. The answer might be as simple as there is no model, since it isn't created yet - but I thought perhaps it's been New'ed up but not saved and there might be a way to access it?

**Comment by danharrin** [comment]
Yeah you are correct when saying that the model doesnt exist yet. What are you trying to do?

**Reply by intrepidws** [reply]
The model is for a badge (think like one you'd wear to Laracon or something similar). I have a live preview of the badge that is generated as you make edits to it.

Is there a way for me to access the data from the other fields during the create process, from within the custom field?

**Reply by danharrin** [reply]
Yes, from within the view you can try `$evaluate(fn ($get) => $get('field_name'))`, as you would in any other field method

**Reply by intrepidws** [reply]
Thanks Dan.
